components/scream/src/physics/spa/atmosphere_prescribed_aerosol.cpp

```diff
@@ -0,0 +1,193 @@
+
+  Real* mem = reinterpret_cast<Real*>(buffer_manager.get_memory());
+
+  // 1d scalar views
+  m_buffer.ps_src = decltype(m_buffer.ps_src)(mem, m_num_cols);
+  mem += m_buffer.ps_src.size();
+
+  Spack* s_mem = reinterpret_cast<Spack*>(mem);
+
+  // 2d packed views
+  const Int num_mid_packs    = ekat::npack<Spack>(m_num_levs);
+  const Int num_int_packs = ekat::npack<Spack>(m_num_levs+1);
+
+  m_buffer.p_mid_src = decltype(m_buffer.p_mid_src)(s_mem, m_num_cols, num_mid_packs);
+  s_mem += m_buffer.p_mid_src.size();
+  m_buffer.ccn3_src = decltype(m_buffer.ccn3_src)(s_mem, m_num_cols, num_mid_packs);
+  s_mem += m_buffer.ccn3_src.size();
+
+  // WSM data
+  m_buffer.wsm_data = s_mem;
+
+  // Compute workspace manager size to check used memory
+  // vs. requested memory
+  const auto policy  = ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(m_num_cols, num_mid_packs);
+  const int wsm_size = WSM::get_total_bytes_needed(num_mid_packs, 3, policy)/sizeof(Spack);
+  s_mem += wsm_size;
+
+  int used_mem = (reinterpret_cast<Real*>(s_mem) - buffer_manager.get_memory())*sizeof(Real);
+  EKAT_REQUIRE_MSG(used_mem==requested_buffer_size_in_bytes(), "Error! Used memory != requested memory for SPA.");
+}
+
+// =========================================================================================
+void SPA::initialize_impl (const util::TimeStamp& /* t0 */)
+{
+  // Initialize Time Data
+  auto ts = timestamp();
+  SPATimeState.current_month = ts.get_months();
+  SPATimeState.t_beg_month = util::julian_day(ts.get_years(),ts.get_months(),0,0);
+  SPATimeState.days_this_month = (Real)ts.get_dpm();
+
+  // Initialize SPA input data
+  initialize_spa_impl();
+}
+
+// =========================================================================================
+void SPA::run_impl (const Real dt)
+{
+  /* Gather time and state information for interpolation */
+  auto ts = timestamp();
+  /* Update time data if the month has changed */
+  SPATimeState.t_now = ts.get_julian_day();
+  if (ts.get_months() != SPATimeState.current_month) {
+    SPATimeState.current_month = ts.get_months();
+    SPATimeState.t_beg_month = util::julian_day(ts.get_years(),ts.get_months(),0,0);
+    SPATimeState.days_this_month = (Real)ts.get_dpm();
+  }
+
+  // Call the main SPA routine to get interpolated aerosol forcings.
+  SPAFunc::spa_main(SPATimeState, SPAPressureState,SPAData_start,SPAData_end,SPAData_out,m_num_cols,m_num_levs,m_nswbands,m_nlwbands);
+
+  // Advance current timestamp.
+  ts += dt;
+  for (auto& f : m_spa_fields_out) {
+    f.second.get_header().get_tracking().update_time_stamp(ts);
+  }
+}
+
+// =========================================================================================
+void SPA::finalize_impl()
+{
+  // Do nothing
+}
+
+void SPA::set_required_field_impl (const Field<const Real>& f) {
+
+  const auto& name = f.get_header().get_identifier().name();
+  m_spa_fields_in.emplace(name,f);
```

Comment:
This method will have to be removed, following what's done in #1165 , which I think will go in before this PR, given it's been approved and passed testing, while I'm gonna request changes here, :)